### PR TITLE
[MB-10712] Payment request cache fixes and optimizations

### PR DESCRIPTION
**Pull request by reggieriser**
## Summary

This ticket addresses a few things with payment requests:

- Param lookup caching can result in incorrect pricing in some scenarios.  See this [prior ticket](https://dp3.atlassian.net/browse/MB-9497) as well as [this document](https://docs.google.com/document/d/1dkSVly3TC8GxpqOEutDyUBvQKTcmQHxQ1n8c1ni-wPY/edit#heading=h.92o2nzwiokvt) for more details about this issue.  I ended up just turning the lookup cache off as discussed in [this Slack thread](https://ustcdp3.slack.com/archives/CP6F568DC/p1633465395092500) since the cache seemed to have a pretty minor effect on the number of queries executed.  I added tests (leveraged from a [previous draft PR](https://github.com/transcom/mymove/pull/7524)) to demonstrate the scenario and how turning off this portion of the cache fixes it.
- As part of my evaluation, I managed to find a few places where I could optimize existing queries (by preloading associations, reusing values already looked up, etc.).  In my local testing with a representative payment request, it looked to result in about 25% fewer queries overall.
- I kept the param map caching as is, but I did turn off an out-of-date (and seemingly not that advantageous) manual mapping previously used when the cache was off.

With the benefit of hindsight, I think we could rethink the cache and some of the payment request code to make it more efficient and easier to navigate.  But that's a bigger effort that's outside the scope of this bug ticket.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

There should be no visible changes as a result of this PR.  Ensure tests are passing.  But you can also try out some payment requests and verify that they still succeed and the pricing seems accurate.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go [hunk 1]
```diff
@@ -466,55 +467,13 @@ func (s *ServiceItemParamKeyData) setLookup(appCtx appcontext.AppContext, servic
...
-			case models.ReServiceCodeDUPK:
-				return false, nil
-			}
-		case models.ServiceItemParamNameServicesScheduleDest:
-			switch serviceItemCode {
-			case models.ReServiceCodeDPK, models.ReServiceCodeDNPK:
-				return false, nil
-			}
-		}
+		// We used to turn some (but not nearly all) lookups on and off with a big switch here if the cache was not
```
Comment (LeDeep) [inline pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go:470]: this comment is helpful. Thanks for adding it

**Review by LeDeep — APPROVED** [review]
LGTM! tests are all passing

**Review by mlm55 — APPROVED** [review]
Thanks for all the explanatory comments!

**Comment by reggieriser** [comment]
@Mergifyio refresh

**Comment by reggieriser** [comment]
@Mergifyio update

**Pull request merged**